services/codecov/codecov.service.js

```diff
@@ -100,40 +99,33 @@ module.exports = class Codecov extends BaseJsonService {
     }
   }
 
-  async fetch({ vcsName, user, repo, branch, token }) {
-    // Codecov Docs: https://docs.codecov.io/reference#section-get-a-single-repository
-    let url = `https://codecov.io/api/${vcsName}/${user}/${repo}`
-    if (branch) {
-      url += `/branches/${branch}`
-    }
-    const options = {}
-    if (token) {
-      options.headers = {
-        Authorization: `token ${token}`,
-      }
-    }
-    return this._requestJson({
+  async fetch({ vcsName, user, repo, branch, token, flag }) {
+    const url = `https://codecov.io/${vcsName}/${user}/${repo}${
+      branch ? `/branches/${branch}` : ''
+    }/graph/badge.svg`
+    const queryParams = [flag && `flag=${flag}`, token && `token=${token}`]
+      .filter(Boolean)
+      .join('&')
+    return this._requestSvg({
       schema,
-      options,
-      url,
-      errorMessages: {
-        401: 'not authorized to access repository',
-        404: 'repository not found',
-      },
+      valueMatcher: svgValueMatcher,
+      url: `${url}?${queryParams}`,
```

Comment:
Inline suggestion disclaimer that prettier/eslint/etc. may want to change parts of this
```suggestion
      url,
      qs: { flag, token }
```